design/design-doc-operations.md

```diff
@@ -0,0 +1,501 @@
+# Day Two Operations
+
+* Owner: Nic Cope (@negz)
+* Reviewer: Hasan Turken (@turkenh)
+* Status: Draft
+
+## Background
+
+Crossplane has three main components:
+
+* Composition - an engine used to build opinionated self-service APIs (composite
+  resources, or XRs)
+* Managed resources (MRs) - a library of Kubernetes APIs for managing popular
+  cloud resources
+* The package manager - a tool for declaratively extending Crossplane with new
+  functionality
+
+Most organizations use these components to manage cloud infrastructure. Platform
+engineering teams use Composition to build opinionated, self-service APIs that
+are abstractions of one or more kinds of MR.
+
+With [Crossplane v2][1] Crossplane will be simpler and better suited to managing
+applications, not just the infrastructure they depend on. Notably XRs will be
+able to compose any Kubernetes resource - not only Crossplane MRs.
+
+This means you can use Crossplane v2 to build a great control plane for basic
+lifecycle management of applications and infrastructure. By basic lifecycle
+management I mean the control plane can create, update, and delete (CRUD) things.
+
+On the other hand, it's hard today to use Crossplane to build a control plane
+that can handle "day two" operations. Things like rolling upgrades, scheduling,
+backups, restores, or misconfiguration detection and remediation. Essentially
+anything beyond CRUD operations. Typically organizations must break out of the
+Crossplane ecosystem and use a tool like [kubebuilder][2] to build a bespoke
+controller for their day two operations.
+
+## Goals
+
+The goal of this design is to enable you to use Crossplane to build a control
+plane that handles your day two operations.
+
+Examples of day two operations include:
+
+* A rolling update of a fleet of Kubernetes cluster XRs
+* Scheduling App XRs to Kubernetes cluster XRs
+* Detecting and pausing MRs stuck in a tight reconcile loop
+* Backing up databases when you delete an RDS instance MR
+
+This is far from an exhaustive list - day two operations are pretty open ended.
+
+The design should make it possible for you to:
+
+* Use a function pipeline to build controllers that handle day two operations.
+* Package and depend on day two operation controllers.
+* Trigger a day two operation by watching arbitrary KRM resources.
+* Trigger a day two operation on a regular interval.
```

Comment:
Is there thoughts on supporting a trigger based on a HTTP request? I know this starts to potentially be out of scope of Crossplane however I feel like the only real other way to get a similar use case is to use like Kubernetes events, or some other CRD..